miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultBoxService.java

```diff
@@ -213,11 +248,21 @@ public BoxableView getBoxableView(BoxableId id) throws IOException {
     return boxStore.listAllBoxSizes();
   }
 
+  @Override
+  public BoxSize getSize(long id) throws IOException {
+    return boxStore.getSizeById(id);
+  }
+
   @Override
   public Collection<BoxUse> listUses() throws IOException {
     return boxStore.listAllBoxUses();
   }
 
+  @Override
+  public BoxUse getUse(long id) throws IOException {
+    return boxStore.getUseById(id);
+  }
+
   @Override
   public long save(Box box) throws IOException {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 32 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)